Assets/ZSSRichTextEditor.js

```diff
@@ -1728,33 +1734,42 @@ ZSSField.prototype.handleFocusEvent = function(e) {
 };
 
 ZSSField.prototype.handleKeyDownEvent = function(e) {
-    
+
     var wasEnterPressed = (e.keyCode == '13');
     
-    if(wasEnterPressed) {
+    if (this.isComposing) {
+        e.stopPropagation();
+    } else if (wasEnterPressed) {
         ZSSEditor.formatNewLine(e);
-    } else {
+    } else if (ZSSEditor.closerParentNode() == this.wrappedDomNode()) {
         // IMPORTANT: without this code, we can have text written outside of paragraphs...
         //
-        if (ZSSEditor.closerParentNode() == this.wrappedDomNode()) {
-            document.execCommand('formatBlock', false, 'p');
-        }
+        document.execCommand('formatBlock', false, 'p');
     }
 };
 
 ZSSField.prototype.handleInputEvent = function(e) {
-    
-    // IMPORTANT: we want the placeholder to come up if there's no text, so we clear the field if
-    // there's no real content in it.  It's important to do this here and not on keyDown or keyUp
-    // as the field could become empty because of a cut or paste operation as well as a key press.
-    // This event takes care of all cases.
-    //
-    this.emptyFieldIfNoContentsAndRefreshPlaceholderColor();
-    
-    var joinedArguments = ZSSEditor.getJoinedFocusedFieldIdAndCaretArguments();
+    if (!this.isComposing) {
```

Comment:
Maybe a question of personal style but it would be nicer to test for truth and return instead of doing the work inside the conditional block.  Not a blocker by any stretch, just a thought.  
` if (this.isComposing ) { return;  } `